### ISSUE TITLE
error when quantize activations to 1 bit.

### ISSUE THREAD

**Issue by ky-du** [zzzxxxttt/pytorch_DoReFaNet#8]
According to the paper, when A_bit is set to 1, activations should also be quantized with q_{k}, rather than Sign() function. 

Experiment shows that the acccuracy of  1-1 bit cifar_resnet only achieves 60%. After replacing Sign() with q_{k}, everything seems normal.

But I have not figured out why Sign() fails, please tell me if you have any idea :)

**Comment by zzzxxxttt** [comment]
So you replaced q_(1) with sign() and the performance is much lower? I guess that's because sign() is not an STE. q_(k) behaves like a linear function during backpropagation, but sign() returns zero gradients.

**Comment by ky-du** [comment]
> So you replaced q_(1) with sign() and the performance is much lower? I guess that's because sign() is not an STE. q_(k) behaves like a linear function during backpropagation, but sign() returns zero gradients.

Oh I see, thank you!
